views-htmx/src/main/java/io/micronaut/views/htmx/http/HtmxResponse.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.htmx.http;
+
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.views.ModelAndView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * HTMX Response.
+ * @author Sergio del Amo
+ * @since 6.0.0
+ * @param <T> The model type
+ */
+@Experimental
+public final class HtmxResponse<T> {
```

Comment:
is the plan to enable more builder methods to set common HTMX response headers?

Reply:
I thought about that. However, I am not sure if it is necessary since you can do: 

```java
HttpResponse.ok(HttpResponse.builder()
    .modelAndView()
    .modelAndView()
    .build())
     .header(HtmxResponseHeaders.HX_REFRESH, StringUtils.TRUE);
```